### PR TITLE
improvement: adding tests to improve mutation testing coverage - PR1

### PR DESCRIPTION
**Pull request by KiranVarma02**
Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-1645

### Change description ###

adding tests to improve mutation testing coverage

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[* ] No
```

### CODE REVIEW

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    private static final Long CASE_ID = 12345L;
+
+    @Autowired
+    private C2UploadedEmailContentProvider c2UploadedEmailContentProvider;
+
+    @PostConstruct
+    void setField() {
+        ReflectionTestUtils.setField(c2UploadedEmailContentProvider, "uiBaseUrl", BASE_URL);
+    }
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProviderTest.java:32]: alternatively you can annotate class with
@TestPropertySource(properties = {"ccd.ui.base.url=" + BASE_URL})

File: service/src/test/java/uk/gov/hmcts/reform/fpl/request/RequestDataTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+public class RequestDataTest {
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/request/RequestDataTest.java:15]: minor: public is not need, here and in all test method

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/UserDetailsServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import uk.gov.hmcts.reform.fpl.request.RequestData;
+import uk.gov.hmcts.reform.idam.client.IdamApi;
+import uk.gov.hmcts.reform.idam.client.models.UserInfo;
+
+import static org.apache.commons.lang.RandomStringUtils.randomAlphanumeric;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+public class UserDetailsServiceTest {
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/UserDetailsServiceTest.java:17]: minor: public not needed here

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Test
+    void shouldReturnExpectedMapWithGivenCaseDetails() {
+        Map<String, Object> expectedMap = ImmutableMap.<String, Object>builder()
+            .put("caseUrl", buildCaseUrl("12345"))
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProviderTest.java:37]: there is constant for caseid at the top of file

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProviderTest.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            .put("reference","12345")
+            .build();
+
+        assertThat(c2UploadedEmailContentProvider.buildC2UploadNotification(populatedCaseDetails())).isEqualTo(expectedMap);
+    }
+
+    @Test
+    void shouldReturnExpectedPbaPaymentNotTakenNotification() {
+        Map<String, Object> expectedMap = ImmutableMap.<String, Object>builder()
+            .put("caseUrl", buildCaseUrl("12345"))
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProviderTest.java:50]: there is constant for caseid

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProviderTest.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+            .build();
+
+        assertThat(c2UploadedEmailContentProvider.buildC2UploadPbaPaymentNotTakenNotification(createCase())).isEqualTo(expectedMap);
+    }
+
+    String buildCaseUrl(String caseId) {
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProviderTest.java:57]: private static
In this method we pass caseId, next method assumes that CASE_ID is used, should be consistent

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProviderTest.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        return formatCaseUrl(BASE_URL, Long.parseLong(caseId));
+    }
+
+    private CaseDetails createCase() {
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProviderTest.java:61]: can be static

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+    @Autowired
+    private CaseManagementOrderEmailContentProvider caseManagementOrderEmailContentProvider;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @PostConstruct
+    void setField() {
+        ReflectionTestUtils.setField(caseManagementOrderEmailContentProvider, "uiBaseUrl", BASE_URL);
+    }
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java:51]: you can use:
@TestPropertySource(properties = {"ccd.ui.base.url=" + BASE_URL})
at the class level

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+
+    @Test
+    void shouldBuildCMOIssuedCaseLinkNotificationParameters() {
+        Map<String, Object> expectedMap = ImmutableMap.<String, Object>builder()
+            .put("localAuthorityNameOrRepresentativeFullName", "testName")
+            .put("caseUrl", buildCaseUrl("12345"))
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java:57]: there is constant for it

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java [hunk 10]
```diff
@@ -0,0 +1,175 @@
...
+    private ObjectMapper objectMapper;
+
+    @PostConstruct
+    void setField() {
+        ReflectionTestUtils.setField(caseManagementOrderEmailContentProvider, "uiBaseUrl", BASE_URL);
+    }
+
+    @Test
+    void shouldBuildCMOIssuedCaseLinkNotificationParameters() {
+        Map<String, Object> expectedMap = ImmutableMap.<String, Object>builder()
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java:55]: expectedParameters is better name

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+
+        assertThat(caseManagementOrderEmailContentProvider
+            .buildCMOIssuedCaseLinkNotificationParameters(createCase(), "testName"))
+            .isEqualTo(expectedMap);
+    }
+
+    @Test
+    void shouldBuildCMOIssuedWithoutDocumentLinkNotificationParameters() {
+
+        Map<String, Object> expectedMap = ImmutableMap.<String, Object>builder()
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java:70]: expectedParamters

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+
+    @Test
+    void shouldBuildCMOIssuedWithDocumentLinkNotificationParameters() throws JsonProcessingException {
+
+        byte[] documentContentAsByte = nextBytes(20);
+        String documentContent = new String(Base64.encodeBase64(documentContentAsByte), ISO_8859_1);
+
+        JSONObject expectedDocumentLink = new JSONObject().put("file", documentContent);
+
+        Map<String, Object> expectedMap = ImmutableMap.<String, Object>builder()
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java:90]: expectedParameters

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+            .put("link_to_document", expectedDocumentLink)
+            .put("reference", "12345")
+            .build();
+
+        String expected = objectMapper.writeValueAsString(expectedMap);
+        String actual = objectMapper.writeValueAsString(
+            caseManagementOrderEmailContentProvider.buildCMOIssuedDocumentLinkNotificationParameters(
+                createCase(), "testName", documentContentAsByte));
+
+        JSONAssert.assertEquals(expected, actual, true);
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java:103]: Map<String, Object> actualParameters =
            caseManagementOrderEmailContentProvider.buildCMOIssuedDocumentLinkNotificationParameters(
                createCase(), "testName", documentContentAsByte);

        AssertionHelper.assertEquals(actualParameters, expectedParameters);

then remove ObjectMapper and throws JsonProcessingException

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+            .put("subjectLineWithHearingDate", "lastName, 11")
+            .put("reference", "12345")
+            .build();
+
+        assertThat(caseManagementOrderEmailContentProvider
+            .buildCMOReadyForJudgeReviewNotificationParameters(createCase())).isEqualTo(expectedParameters);
+    }
+
+    String buildCaseUrl() {
+        return formatCaseUrl(BASE_URL, CaseManagementOrderEmailContentProviderTest.CASE_ID);
```
Comment (tomaszpowroznik) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java:140]: just CASE_ID, it is local constant

Comment (KiranVarma02) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java:140]: true, intellij did that :)

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProviderTest.java [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.service.email.content.AbstractEmailContentProviderTest.BASE_URL;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.populatedCaseDetails;
+import static uk.gov.hmcts.reform.fpl.utils.EmailNotificationHelper.formatCaseUrl;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = { JacksonAutoConfiguration.class, C2UploadedEmailContentProvider.class })
+@TestPropertySource(properties = {"ccd.ui.base.url=" + BASE_URL})
```
Comment (Gogetter) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/C2UploadedEmailContentProviderTest.java:22]: interesting... curious question should this and related tests be integration tests? 🤔 @tomaszpowroznik / @akomisarek

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+    @Test
+    void shouldBuildCMOIssuedWithoutDocumentLinkNotificationExpectedParameters() {
+
+        Map<String, Object> expectedParameters = ImmutableMap.<String, Object>builder()
+            .put("cafcassOrRespondentName", "testName")
+            .put("caseUrl", buildCaseUrl())
+            .put("subjectLineWithHearingDate", "lastName, 11")
+            .put("reference", "12345")
```
Comment (Gogetter) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/email/content/CaseManagementOrderEmailContentProviderTest.java:65]: I think we might want to have these as constants as they are now in two different methods?

**Pull request merged**